readthedocs/config/config.py

```diff
@@ -670,6 +671,19 @@ class BuildConfigV2(BuildConfigBase):
         'dirhtml': 'sphinx_htmldir',
         'singlehtml': 'sphinx_singlehtml',
     }
+    valid_os = ['ubuntu-20.04']
+    valid_languages = {
```

Comment:
I think these should be moved to settings/base.py? Not sure what other metadata we need to attach to each entry.

Reply:
I don't have a strong opinion about where this data should leave. However, it may make sense to put in `settings/base.py` to keep the same pattern that we already have, and considering that it's something we will need to upgrade over time to add more supported tools/versions.

Note that we will require a mapping between the version specified by the user (without the patch number) and the final version installed by `asdf`.

Examples: the user specifies `3.6` and we have cached `3.6.11` we should use that one, even if `3.6.15` is already released. At some point, we can upload the pre-compiled `3.6.15` version to our cache and change this mapping to install the newer one.